### PR TITLE
Jvm version checks

### PR DESCRIPTION
**Pull request by nagarev**
## Description

Added preflight checks to stop node from running when Java Version is not supported.

Java version check can be avoided if the `--skip-java-check` flag is set.

## Motivation and Context

Some people use jre16 or alike for running rskj node. We only support LTS versions 8 and 11, so we should restrict in the code versions of java we support.

## How Has This Been Tested?

Using Unit Tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    private static void checkSupportedJavaVersion() {
+        String javaVersion = SystemUtils.getPropertyValue("java.version");
+
+        if (javaVersion == null) {
+            throw new RuntimeException("Unable to detect Java version");
+        }
+
+        if (Arrays.stream(SUPPORTED_JAVA_VERSIONS).noneMatch(v -> isCompatibleVersion(javaVersion, v))) {
+            logger.error("Invalid Java version detected");
+            throw new RuntimeException(String.format("Invalid Java Version '%s'. Supported versions: %s", javaVersion, String.join(", ", SUPPORTED_JAVA_VERSIONS)));
```
Comment (patogallaiovlabs) [inline rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java:28]: Why diff msgs? We can reuse the same msg for log/exception.

File: rskj-core/src/main/java/org/ethereum/config/SystemProperties.java [hunk 2]
```diff
@@ -88,6 +88,8 @@
 
     public static final String PROPERTY_PRINT_SYSTEM_INFO = "system.printInfo";
 
+    public static final String PROPERTY_PREFLIGHT_CHECKS_SKIP_JAVA_VERSION_CHECK = "preflight.checks.skip.java.version.check";
```
Comment (patogallaiovlabs) [inline rskj-core/src/main/java/org/ethereum/config/SystemProperties.java:91]: can we shorten the prop name? like "system.checkJavaVersion"?

File: rskj-core/src/test/java/co/rsk/util/PreflightChecksUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void runChecks_receivesSkipJavaCheck_skipsJavaChecks() {
+        String[] args = {"--skip-java-check"};
+
+        RskTestContext rskContext = new RskTestContext(args);
+
+        PreflightChecksUtils.runChecks(rskContext);
```
Comment (patogallaiovlabs) [inline rskj-core/src/test/java/co/rsk/util/PreflightChecksUtilsTest.java:30]: no assertions, how do we know that it skipped?

**Review by patogallaiovlabs — CHANGES_REQUESTED** [review]

File: rskj-core/src/test/java/co/rsk/util/PreflightChecksUtilsTest.java (hunk 3 above)
Comment (nagarev) [reply rskj-core/src/test/java/co/rsk/util/PreflightChecksUtilsTest.java]: Yeah, it was updated and refactored in a new class.

File: rskj-core/src/main/java/org/ethereum/config/SystemProperties.java (hunk 2 above)
Comment (nagarev) [reply rskj-core/src/main/java/org/ethereum/config/SystemProperties.java:91]: Yes, I think we can.

File: rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java (hunk 1 above)
Comment (nagarev) [reply rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java]: I've updated that so we send the same message. 💪🏼

File: rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+public class PreflightChecksUtils {
+    private static final Logger logger = LoggerFactory.getLogger(PreflightChecksUtils.class);
+
+    private static final String[] SUPPORTED_JAVA_VERSIONS = {"1.8", "11"};
+
+    private static void checkSupportedJavaVersion() {
+        String javaVersion = SystemUtils.getPropertyValue("java.version");
+
+        if (javaVersion == null) {
```
[the remark this answers is not in the document]
Comment (nagarev) [reply rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java]: I should not, but I added the check to be in the "safe" side i think.

File: rskj-core/src/main/java/org/ethereum/config/SystemProperties.java (hunk 2 above)
Comment (nagarev) [reply rskj-core/src/main/java/org/ethereum/config/SystemProperties.java]: Done!

File: rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (patogallaiovlabs) [reply rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java:29]: didn't we agree on removing this check? 😛

[the remark this answers is not in the document]
Comment (nagarev) [reply rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java]: Done!

File: rskj-core/src/main/java/org/ethereum/config/SystemProperties.java [hunk 5]
```diff
@@ -88,6 +88,8 @@
 
     public static final String PROPERTY_PRINT_SYSTEM_INFO = "system.printInfo";
 
+    public static final String PROPERTY_PREFLIGHT_CHECKS_SKIP_JAVA_VERSION_CHECK = "system.checkJavaVersion";
```
Comment (patogallaiovlabs) [inline rskj-core/src/main/java/org/ethereum/config/SystemProperties.java:91]: the name of the const is still long and diff from property

File: rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import co.rsk.config.NodeCliFlags;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+
+/**
+ * Created by Nazaret García on 21/01/2021
```
Comment (patogallaiovlabs) [inline rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java:13]: Add a comment explaining briefly what it does.

Comment (nagarev) [reply rskj-core/src/main/java/co/rsk/util/PreflightChecksUtils.java:13]: Done!

File: rskj-core/src/main/java/org/ethereum/config/SystemProperties.java (hunk 5 above)
Comment (nagarev) [reply rskj-core/src/main/java/org/ethereum/config/SystemProperties.java]: Done!

**Comment by nagarev** [comment]
pipeline: run

**Comment by nagarev** [comment]
pipeline:run

**Review by patogallaiovlabs — APPROVED** [review]
LGTM

**Pull request closed without merging**